servers/visual_server.cpp

```diff
@@ -333,6 +333,10 @@ RID VisualServer::get_white_texture() {
 // Resulting 2D vector in range [-1, 1]
 // See http://jcgt.org/published/0003/02/01/ for details
 Vector2 VisualServer::norm_to_oct(const Vector3 v) {
+	if (Math::is_equal_approx(v.length(), 0)) {
+		return Vector2();
+	}
+
```

Comment:
I think we could just do a warning if the value is not normalized?

something like:
https://github.com/godotengine/godot/blob/a544e77822b40b39198f92ee3346bbfe7921ac41/core/math/vector3.h#L441-L443

We could error with a message that says "Octahedral compression can only be used with normalized values. Please use normalized normal values, or disable Octahedral compression"

Reply:
It is not clear that spamming per vertex is the proper warning.

Reply:
Fair enough, could probably just be ERR_WARN_ONCE

Reply:
I don't want the chance they make it to the renderer supplying them to the GPU would be bad, especially since a lot of us just ignore warnings

Reply:
oh but I can put an ERR_WARN_ONCE in the check too

Reply:
Out of curiosity, is it common to have zero length normal vector (do they serve a purpose at all?)

I think the warning is a good idea, if we look at Godot's behavior historically, we technically supported vectors of any length for normal/tangent (including zero length) as I don't think we did any normalization of normal vectors on mesh import that (I am aware of) - so if we want to stay close to the same behavior we had before perhaps it's a good idea to also return a Vector2(0, 0) in this scenario? It would decode to Vector3(0, 0, 1) but I guess that's better than a NaN and the user would get a warning about it

Reply:
You can have an error that returns a value as well. Maybe ERR_WARN_ONCE_V?